platform/posix/transport/src/openssl_posix.c

```diff
@@ -535,7 +541,7 @@ OpensslStatus_t Openssl_Connect( NetworkContext_t * pNetworkContext,
     }
 
     /* Clean up on error. */
-    if( ( returnStatus != OPENSSL_SUCCESS ) && ( sslStatus == 0 ) )
+    if( ( returnStatus != OPENSSL_SUCCESS ) && ( sslObjectCreated == 1u ) )
```

Comment:
Can you instead check that pNetworkContext->pSsl instead of the sslObjectCreated?

Reply:
That logic doesn't work as the input parameter of `pNetworkContext` can have a non-NULL value for `pSsl`, and thus, we want to free only if _we_ created the object

Reply:
Oh then wouldn't that be an invalid network context anyways?

Reply:
We cannot free a `pNetworkContext->pSsl` that has its memory default initialized by the caller's stack (which results in a SegFault!)